docs/cri/registry.md

```diff
@@ -41,6 +41,39 @@ not specified by Kubernetes via CRI.
 
 After modifying this config, you need to restart the `containerd` service.
 
+### Configure Docker Hub Credentials Example
+
+You may want to configure Docker Hub credentials to pull images from Docker Hub.
+This allows you to pull public and private images from Docker Hub without hitting rate limits.
+
+To configure Docker Hub credentials, create/modify `/etc/containerd/certs.d/registry-1.docker.io/hosts.toml` as follows:
```

Comment:
Creating this `hosts.toml` file is not necessary as the below configuration is already the default.

Reply:
Dear, I can't solve this issue: ctr image pull docker.io/user/image:tag
docker.io/israfilbat22/insight-dashboard-service:desco-v1: resolving      |--------------------------------------| 
elapsed: 1.9 s                                             total:   0.0 B (0.0 B/s)                                         
INFO[0002] trying next host                              error="pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed" host=registry-1.docker.io
ctr: failed to resolve reference "docker.io/user/image:tag": pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed....




here my config.toml file: 

cat /etc/containerd/config.toml 
version = 2
[plugins."io.containerd.grpc.v1.cri".containerd]
default_runtime_name = "runc"

[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
runtime_type = "io.containerd.runc.v2"

[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
SystemdCgroup = true

  [plugins."io.containerd.grpc.v1.cri".registry.mirrors]
    [plugins."io.containerd.grpc.v1.cri".registry.mirrors."docker.io"]
      endpoint = ["https://registry-1.docker.io"]
  [plugins."io.containerd.grpc.v1.cri".registry.configs]
    [plugins."io.containerd.grpc.v1.cri".registry.configs."docker.io".auth]
      username = "weio"
      password = "sa;dfha"


Note: When i user and password use in command line then ctr pull command working good but when i user and password put into config.toml then doesn't work. 

ctr image pull --user 'user:password' docker.io/user/image:tag   -----its working
ctr image pull  docker.io/user/image:tag   ---- its not working

i want to use " ctr image pull/push docker.io/user/image:tag "  credential use from config.toml file

please suggest me.

Reply:
Try using the registry endpoint `[plugins."io.containerd.grpc.v1.cri".registry.configs."registry-1.docker.io".auth]`

Reply:
updated:  [plugins."io.containerd.grpc.v1.cri".registry.configs."registry-1.docker.io".auth]
      username = "@@@@"
      password = "#######"

use this but same error show

here my full config.toml:
cat /etc/containerd/config.toml
version = 2
[plugins."io.containerd.grpc.v1.cri".containerd]
default_runtime_name = "runc"
[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
runtime_type = "io.containerd.runc.v2"
[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc.options]
SystemdCgroup = true

  [plugins."io.containerd.grpc.v1.cri".registry.mirrors]
    [plugins."io.containerd.grpc.v1.cri".registry.mirrors."docker.io"]
      endpoint = ["https://registry-1.docker.io"]
  [plugins."io.containerd.grpc.v1.cri".registry.configs]
    [plugins."io.containerd.grpc.v1.cri".registry.configs."registry-1.docker.io".auth]
      username = "@@@@@"
      password = "########"

Reply:
Discussion is a better place to help with that

Reply:
The configuration below seems to be a specific example for setting up `gcr.io` registries. It mentions a docker host configuration but in a different context.

I was working in Amazon Linux AMIs with `containerd` 1.6.x and needed to create a `hosts.toml` file.

Should the example contents of `hosts.toml` get moved up to `config.toml`?

Reply:
Its not currently relevant to the credential configuration, including it is not necessary.

Reply:
Ah, I see. That makes sense.  I removed the `hosts.toml` bit.